tests/splines/non_periodic_spline_builder.cpp

```diff
@@ -117,28 +120,35 @@ TEST(NonPeriodicSplineBuilderTest, Identity)
             spline_builder(interpolation_domain);
 
     // 5. Allocate and fill a chunk over the interpolation domain
-    ddc::Chunk yvals(interpolation_domain, ddc::KokkosAllocator<double, Kokkos::HostSpace>());
+    ddc::Chunk yvals_alloc(interpolation_domain, ddc::KokkosAllocator<double, memory_space>());
+    ddc::ChunkSpan const yvals(yvals_alloc.span_view());
     evaluator_type evaluator(interpolation_domain);
-    evaluator(yvals.span_view());
+    ddc::parallel_for_each(
+            yvals.domain(),
+            KOKKOS_LAMBDA(IndexX const ix) { yvals(ix) = evaluator(ddc::coordinate(ix)); });
 
     int constexpr shift = s_degree_x % 2; // shift = 0 for even order, 1 for odd order
-    ddc::Chunk derivs_lhs_alloc(derivs_domain, ddc::HostAllocator<double>());
+    ddc::Chunk derivs_lhs_alloc(derivs_domain, ddc::KokkosAllocator<double, memory_space>());
     ddc::ChunkSpan derivs_lhs = derivs_lhs_alloc.span_view();
     if (s_bcl == ddc::BoundCond::HERMITE) {
-        for (ddc::DiscreteElement<ddc::Deriv<DimX>> const ii : derivs_domain) {
-            derivs_lhs(ii) = evaluator.deriv(x0, ii - derivs_domain.front() + shift);
-        }
+        ddc::parallel_for_each(
+                derivs_domain,
```

Comment:
Should the exec_space be passed as an argument here in case you want to test Host? Or do you not care as you can simply compile without GPU support?

Reply:
The test is made to run on the default execution space so it should be fine if we compile for CPU or GPU. I tend to prefer to pass explicitly the execution space object so I am ok to add it.

Reply:
Yes, my question was about if you changed the first line of the test from:
```cpp
using execution_space = Kokkos::DefaultExecutionSpace;
```
back to:
```cpp
using execution_space = Kokkos::DefaultHostExecutionSpace;
```
(e.g for debugging purposes).

> I tend to prefer to pass explicitly the execution space object so I am ok to add it.

Ok. There are a few other similar loops in this PR

Reply:
I can make the changes indeed

Reply:
addressed in 91921a646813b7aaccae7f859e1d9cabf83e3a9d